### ISSUE TITLE
Is there any way to retreive Vendor ID and Product Id from removable devices?

### ISSUE THREAD

**Issue by sushiljic** [samuelcampos/usbdrivedetector#27]
I need to use verify the pendrive by its vendor ID and Product ID for its uniqueness so is there way to get VID and PID?

**Comment by MarkJeronimus** [comment]
Not with this library, but on linux you could execute 'lsusb' and see if the VID:UID is listed (the device and/or mount point will not be listed so there's no way to link anything if there are multiple identical pendrives attached.

**Comment by samuelcampos** [comment]
Do you need to identify the pendrive using the VID and PID specifically? Can't you use the Serial Number instead?

**Comment by sushiljic** [comment]
@samuelcampos 
I need vendor ID and product ID because It can't be changed easily whereas Serial Number can be changed and also It get changed when formatted.
It is good library and It would be best if we can retrieve VID and PID too.

**Comment by marian-m12l** [comment]
> @samuelcampos
> I need vendor ID and product ID because It can't be changed easily whereas Serial Number can be changed and also It get changed when formatted.
> It is good library and It would be best if we can retrieve VID and PID too.

Indeed. When expecting a specific device, one could make sure to find the mount point of that particular device.
Library looks great anyway! :-)

**Comment by Avec112** [comment]
If you already know what USB device that is used one way that is supported is I give the USB device a `Volume Name` and inside the code check for `device.getDeviceName().equalsIgnoreCase("myVolumeName")`. Also the mount point might not be that important since the device has a method which returns the root directory as a file. That would be `device.getRootDirectory()`.

**Comment by ouaibsky** [comment]
maybe it can help you (only jdk21 with preview):

* https://github.com/manuelbl/JavaDoesUSB
